### PR TITLE
[ENH, BUG] Test honest tree performance via quadratic simulation

### PR DESCRIPTION
**Pull request by adam2392**
Fixes #157 

Changes proposed in this pull request:
- Fixes API for calling `n_estimators`
- Adds additional testing towards fixing the honest tree power performance via quadratic simulation

## Before submitting

- [ ] I've read and followed all steps in the [Making a pull request](https://github.com/py-why/pywhy-graphs/blob/main/CONTRIBUTING.md#making-a-pull-request)
    section of the `CONTRIBUTING` docs.
- [ ] I've updated or added any relevant docstrings following the syntax described in the
    [Writing docstrings](https://github.com/py-why/pywhy-graphs/blob/main/CONTRIBUTING.md#writing-docstrings) section of the `CONTRIBUTING` docs.
- [ ] If this PR fixes a bug, I've added a test that will fail without my fix.
- [ ] If this PR adds a new feature, I've added tests that sufficiently cover my new functionality.

## After submitting

- [ ] All GitHub Actions jobs for my pull request have passed.

### CODE REVIEW

File: sktree/tests/test_honest_forest.py [hunk 1]
```diff
@@ -252,3 +255,45 @@ def test_importances(dtype, criterion):
...
+    assert_allclose(np.mean(honestsk_scores), np.mean(honest_scores))
+
+
+def test_honest_forest_with_sklearn_trees_with_power():
+    """Test against regression in power-curves discussed in:
+    https://github.com/neurodata/scikit-tree/pull/157.
+    
+    This unit-test now tests explicitly the power curve.
+    """
+    pass
```
Comment (adam2392) [inline sktree/tests/test_honest_forest.py]: Migrating disc from https://github.com/neurodata/scikit-tree/pull/152#discussion_r1380616968 here.

Can you, @PSSF23 and I discuss to get a small example we can run? 

This is my impression of how the power is computed

File: sktree/tests/test_honest_forest.py [hunk 2]
```diff
@@ -252,3 +255,73 @@ def test_importances(dtype, criterion):
...
+    # clf = HonestForestClassifier(n_estimators=10, random_state=0)
+    # honest_scores = cross_val_score(clf, X, y, cv=5)
+    # print(honest_scores)
+
+    print(honestsk_scores, honest_scores)
+    print(np.mean(honestsk_scores), np.mean(honest_scores))
+    assert_allclose(np.mean(honestsk_scores), np.mean(honest_scores))
+
+
+def test_honest_forest_with_sklearn_trees_with_power():
```
Comment (adam2392) [inline sktree/tests/test_honest_forest.py]: Migrating disc from https://github.com/neurodata/scikit-tree/pull/152#discussion_r1380616968 here.

Can you, @PSSF23 and I discuss to get a small example we can run?

This is my impression of how the power is computed

Comment (sampan501) [reply sktree/tests/test_honest_forest.py]: The only difference here is that you are computing AUC and I'm computing MI. Otherwise, looks right to me

**Review by PSSF23 — COMMENTED** [review]
I added another test on MI to see if the statistic method affects the test results.

File: sktree/tests/test_honest_forest.py (hunk 2 above)
Comment (adam2392) [reply sktree/tests/test_honest_forest.py]: Looks like both MI and AUC have the exact same scores up to `0.05` precision when set with the same random state, so this is leading me to believe there's some issue w/ randomness when running the power-curve simulation, or another bug that we're not thinking of.

**Review by PSSF23 — COMMENTED** [review]
@adam2392 I have doubts on these two lines:

https://github.com/neurodata/scikit-tree/blob/e4728fa9923103c9e219825a46a83b7ba86b483e/sktree/stats/forestht.py#L1071

https://github.com/neurodata/scikit-tree/blob/e4728fa9923103c9e219825a46a83b7ba86b483e/sktree/stats/forestht.py#L1133

Essentially if we set `sample_dataset_per_tree` and `permute_forest_fraction` to `False`, these two lines would result in potential conflicts, or the first line is unnecessary.

**Comment by adam2392** [comment]
> @adam2392 I have doubts on these two lines:
> 
> https://github.com/neurodata/scikit-tree/blob/e4728fa9923103c9e219825a46a83b7ba86b483e/sktree/stats/forestht.py#L1071
> 
> https://github.com/neurodata/scikit-tree/blob/e4728fa9923103c9e219825a46a83b7ba86b483e/sktree/stats/forestht.py#L1133
> 
> Essentially if we set `sample_dataset_per_tree` and `permute_forest_fraction` to `False`, these two lines would result in potential conflicts, or the first line is unnecessary.

The first line can be removed.

Or rather more specifically these two lines: https://github.com/neurodata/scikit-tree/blob/e4728fa9923103c9e219825a46a83b7ba86b483e/sktree/stats/forestht.py#L818-L820

These lines shouldn't affect the issue described in this PR tho.

**Pull request merged**